app/src/main/java/com/humberto/tasky/core/database/entity/ModifiedTaskEntity.kt

```diff
@@ -0,0 +1,19 @@
+package com.humberto.tasky.core.database.entity
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import com.humberto.tasky.core.database.ModificationType
+
+@Entity
+data class ModifiedTaskEntity(
+    @PrimaryKey(autoGenerate = true)
+    val id: Long = 0,
+    val taskId: String,
+    val modificationType: ModificationType,
+    val title: String? = null,
+    val description: String? = null,
+    val time: Long? = null,
+    val remindAt: Long? = null,
+    val isDone: Boolean? = null,
+    val timestamp: Long = System.currentTimeMillis()
```

Comment:
I believe the purpose behind this class is to keep track of the API calls that need to be made to sync the local DB with the remote API. I’ll go based on this assumption with my comment, so if that is not the case, please correct me 😄 

When we need to sync data, it is because the initial API call failed. What we need to do next is save the relevant data and schedule a retry at a later stage. This is waht you’ve done here, but it can be simplified to not have any nullable fields.

```
@Entity
data class TaskPendingSyncEntity(
  @Embedded val task: TaskEntity,
  @PrimaryKey(autoGenerate = false)
  val taskId: String = task.id,
  val userId: String
)
```

Room allows us to embed an existing entity into another (it will flatten it in the table to multiple columns). Now we have all the data we need - we don’t specifically care about what data was modified, and the API requires a complete model regardless of creating and updating agenda items. We also include the user's ID - this is done to preserve these syncs across user login sessions and prevent calling sync calls for another user who has logged into this device.

For delete task syncs, the entity gets simpler:
```
@Entity
data class DeletedTaskSyncEntity(
  @PrimaryKey(autoGenerate = false)
  val taskId: String,
  val userId: String
)
```

And for updating events, a bit more info is required:
```
@Entity
data class EventPendingUpdateSyncEntity(
  @Embedded val event: EventEntity,
  @PrimaryKey(autoGenerate = false)
  val eventId: String = event.id,
  val photosToUploadUris: List<String>,
  val userId: String,
  val isGoing: Boolean,
  val deletedPhotoKeys: List<String>
)
```

You’ll notice that this approach also encapsulates precisely what we need for each sync operation (keeping the code focused and deliberate). The `ModifierTaskEntity` approach is a catch-all solution that can lead to creative workarounds when limitations are reached later in development.

Reply:
Awesome, super valuable, thank you!

Reply:
Kai, one question, for updating Tasks and Reminders where we don't need different fields like Events, how do we track if it's creating or updating with TaskPendingSyncEntity?

Reply:
The table itself does not have a way to distinguish them. Either an `UpdateTaskWorker` or a `CreateTaskWorker` is scheduled, they would have an ID param to retrieve the pending entity. Each worker would then call the appropriate API endpoint to either update or create the task. I had no need to create separate entities since both had the same fields, so I kept it as the same entity for simplicity.

If you would need to differentiate them or if the models change for some reason, then refactoring would, fortunately, be simple - that is why Philipp promotes the architectural style that he does in his videos: to make our lives as developers easier 😄